Pal/Console.Uefi.cs

```diff
@@ -0,0 +1,143 @@
+﻿namespace System
+{
+    public static class Console
+    {
+        static ushort s_consoleAttribute;
+        static ushort s_cursorX;
+        static ushort s_cursorY;
+        static ushort s_windowSizeX;
+        static ushort s_windowSizeY;
+
+        public static unsafe string Title
+        {
+            set
+            {
+            }
+        }
+
+        public static unsafe bool CursorVisible
+        {
+            set
+            {
+                EfiRuntimeHost.SystemTable->ConOut->EnableCursor(EfiRuntimeHost.SystemTable->ConOut, value);
+            }
+        }
+
+        public unsafe static ConsoleColor ForegroundColor
+        {
+            set
+            {
+                s_consoleAttribute = (ushort)value;
+                uint color = s_consoleAttribute;
+                EfiRuntimeHost.SystemTable->ConOut->SetAttribute(EfiRuntimeHost.SystemTable->ConOut, color);
+            }
+        }
+
+
+        static char lastKey = '\0';
+        static ushort lastScanCode;
+        public static unsafe bool KeyAvailable
+        {
+            get
+            {
+                if (lastKey == '\0')
+                {
+                    return true;
+                }
+
+                EFI_INPUT_KEY key;
+                var errorCode = EfiRuntimeHost.SystemTable->ConIn->ReadKeyStroke(EfiRuntimeHost.SystemTable->ConIn, &key);
+                lastKey = (char)key.UnicodeChar;
+                return errorCode == 0;
+            }
+        }
+
+        public static unsafe void Clear()
+        {
+            EfiRuntimeHost.SystemTable->ConOut->ClearScreen(EfiRuntimeHost.SystemTable->ConOut);
+        }
+
+        public static unsafe ConsoleKeyInfo ReadKey(bool intercept)
+        {
+            if (lastKey == '\0')
+            {
+                EFI_INPUT_KEY key;
+                var errorCode = EfiRuntimeHost.SystemTable->ConIn->ReadKeyStroke(EfiRuntimeHost.SystemTable->ConIn, &key);
+                lastKey = (char)key.UnicodeChar;
+                lastScanCode = key.ScanCode;
+            }
```

Comment:
This should probably loop until `ReadKeyStroke` succeeds because we might end up using bogus key data (`ReadKeyStroke` won't block). Might be easier to just delete this - the game is making sure ReadKey is not called unless KeyAvailable returned true.

Reply:
I think you are right on all that console reading stuff. Simplify it.